step-capstone/src/components/Sidebars/HandleClickedTimePoint.js

```diff
@@ -0,0 +1,75 @@
+/* Handling click into empty slot starts HERE */
+// Receives displayDate, returns the date string of displayDate
+const getDisplayDateString = (displayDate) => {
+  let year = displayDate.getFullYear().toString();
+  let month = (displayDate.getMonth() + 1 <10)? "0" + (displayDate.getMonth() + 1).toString() : displayDate.getMonth();
+  let date = (displayDate.getDate() < 10)? "0" + displayDate.getDate()   : displayDate.getDate();;
+  return year + "-" + month + "-" + date +   "T";
+};
+
+// Receives displayItems list, returns a list of empty intervals
+export const getEmptySlots = (displayItems, displayDate) => {
+  let emptySlots = [];
+  const displayDateString = getDisplayDateString(displayDate);
+  if (displayItems.length == 0) {
+    emptySlots.push({ 
+        startDate: new Date(displayDateString + "00:00:00"),
+        endDate: new Date(displayDateString + "23:59:59"),
+    });
+    return emptySlots;
+  } else {
+    emptySlots.push({
+      startDate: new Date(displayDateString + "00:00:00"),
+      endDate: displayItems[0].startDate,
+    });
+  }
+
+  for (var i = 1; i <= displayItems.length - 1; i++) {
+    let prev = displayItems[i - 1];
+    let cur = displayItems[i];
+
+    emptySlots.push({startDate: prev.endDate, endDate: cur.startDate});
+  }
+
+  emptySlots.push({
+    startDate: displayItems[displayItems.length - 1].endDate,
+    endDate: new Date(displayDateString + "23:59:59"),
+  });
+  return emptySlots;
+};
+
+//Receives a list of intervals and returns the interval that the timePoint is in, returns null if there is none.
+const binarySearch = (intervals, timePoint) => {
+  if (intervals.length == 0) {
+    return null;
+  }
+  let low = 0;
+  let high = intervals.length - 1;
+
+  while (low <= high) {
+    let centerIndex = Math.floor((low + high) / 2);
+    let centerInterval = intervals[centerIndex];
+    if (
+      centerInterval.startDate <= timePoint &&
+      timePoint <= centerInterval.endDate
+    ) {
+      return centerInterval;
+    } else if (timePoint < centerInterval.startDate) {
+      high = centerIndex - 1;
+    } else {
+      low = centerIndex + 1;
+    }
+  }
+
+  return null;
+};
+
+/*Return the empty timeRange that users click into by performing binary search on emptySlots array*/
+export const handleClickedTimePoint = (idV, displayDate, emptySlots, displayItems) => {
```

Comment:
I am probably misunderstanding something, but could all this be simplified? If we have displayItems, all we need to do is find where displayItems[i].startDate>clickedTime? That would be the endDate of our empty interval and the prior item's endDate would be our startDate. Why do we need to search through a seperate list of empty slots?

Reply:
You're right, Zach. I can implement it the way you said, I'll just need to handle the case of where the user clicks in a completely empty timeline or the last time slot of the day by adding a fake travelobject at 23:59. When I implemented it, I was just thinking that those base cases were much, but now thinking back, it's not much at all. Okay, I will change this algorithm accordingly to your suggestion. 👍